### DISCUSSION TITLE
slow api responses with large data, compared to populatedb

### DISCUSSION THREAD

**Discussion by nick-chal** [saleor/saleor#7110] [Q&A]
I have a testing server where I have populated the db with around 20k products and 500 categories with many nested subcategories.
But after the import, the response from the graphql api after the import is very slow (>4s in some cases) compared to when I only had the `populatedb` data. Any idea what might be the reason? Does the server specs affect the performance? Or am I missing any optimization configs? 
I am using saleor 3.0

Also another thing to note is the FCPs(initial load times) for the storefront and dashboard are also very high.

If anyone has any ideas or point me to right direction?

**Comment by krzysztofwolski** [comment]
Can you provide an example of the request? ATM we are optimizing 3.0 before the official release, so additional data would help us.

**Reply by nick-chal** [reply]
Especially the queries that require the products or categories take long time to get the response.
For instance in the dashboard where we need the list of categories and some additional data, it is taking about ~4s for the response.
![Screenshot from 2021-04-07 08-19-24](https://user-images.githubusercontent.com/37323052/113802330-3e2ef080-977a-11eb-89d7-dd19f7e3482a.png)

**Reply by krzysztofwolski** [reply]
I'm passing info to the team. BTW Are you using latest 3.0 release?

**Reply by krzysztofwolski** [reply]
https://github.com/mirumee/saleor/pull/7045 has been merged lately, so it should address category performance

**Reply by nick-chal** [reply]
thanks, for the swift reply. really appreciate it.

will try with the latest changes and see if there's any change.
